omnigan/generator.py

```diff
@@ -280,11 +288,14 @@ class SpadeTranslationDict(nn.ModuleDict):
     def __init__(self, opts):
```

Comment:
This is the main class to look at, it does the trick of setting the inner bit value of the spades translation decoder depending on the key that is accessed (f or n) and therefore the translation direction